src/mbgl/tile/geometry_tile_data.cpp

```diff
@@ -173,8 +173,47 @@ static Feature::geometry_type convertGeometry(const GeometryTileFeature& geometr
     return Point<double>();
 }
 
-Feature convertFeature(const GeometryTileFeature& geometryTileFeature, const CanonicalTileID& tileID) {
-    Feature feature { convertGeometry(geometryTileFeature, tileID) };
+static Feature::geometry_type convertGeometryType(const GeometryTileFeature& geometryTileFeature) {
+    GeometryCollection geometries = geometryTileFeature.getGeometries();
```

Comment:
`getGeometries()` ends up processing polygons - which is one of the hits you want to avoid. You might need to add a `GeometryTileFeature.getGeometryType()` method to get the type of the underlying geometry, as opposed to `FeatureType`